### PR TITLE
fix MultipartBody

### PR DESCRIPTION
**Pull request by droplet-js**
it doesn't work for C# Server when set charset='utf8'

### CODE REVIEW

File: lib/src/multipart_body.dart [hunk 1]
```diff
@@ -8,19 +8,17 @@ final _random = Random();
 
 class MultipartBody implements RequestBody {
   final List<Part> parts;
-  MediaType _contentType;
+  MediaType type;
```
Comment (tiagohm) [inline lib/src/multipart_body.dart]: should use "_contentType" as it was before

```suggestion
  MediaType _contentType;
```

File: lib/src/multipart_body.dart [hunk 2]
```diff
@@ -8,19 +8,17 @@ final _random = Random();
...
   String _boundary;
 
   MultipartBody({
     this.parts = const [],
+    this.type = MediaType.multipartMixed,
```
Comment (tiagohm) [inline lib/src/multipart_body.dart]: Should use `multipartFormData` as default value.

```suggestion
    MediaType type = MediaType.multipartFormData,
```

File: lib/src/multipart_body.dart [hunk 3]
```diff
@@ -8,19 +8,17 @@ final _random = Random();
...
     String boundary,
   })  : assert(boundary == null || boundary.isNotEmpty),
+        assert(type != null && type.type == 'multipart'),
         assert(parts != null) {
     _boundary = boundary ?? ('X-RESTIO-${_generateBoundary()}');
-    _contentType = MediaType.multipartFormData.copyWith(
```
Comment (tiagohm) [inline lib/src/multipart_body.dart:20]:
```dart
_contentType = type.copyWith(boundary: _boundary);
```

File: lib/src/multipart_body.dart [hunk 4]
```diff
@@ -39,24 +37,26 @@ class MultipartBody implements RequestBody {
   }
 
   @override
-  MediaType get contentType => _contentType;
+  MediaType get contentType => type.copyWith(boundary: _boundary);
```
Comment (tiagohm) [inline lib/src/multipart_body.dart]:
```suggestion
  MediaType get contentType => _contentType;
```

File: lib/src/multipart_body.dart [hunk 5]
```diff
@@ -39,24 +37,26 @@ class MultipartBody implements RequestBody {
...
   String get boundary => _boundary;
 
   MultipartBody copyWith({
     List<Part> parts,
+    MediaType type,
     String boundary,
   }) {
     return MultipartBody(
       parts: parts ?? this.parts,
+      type: type ?? this.type,
```
Comment (tiagohm) [inline lib/src/multipart_body.dart]:
```suggestion
      type: type ?? _contentType,
```

**Review by tiagohm — CHANGES_REQUESTED** [review]

File: lib/src/multipart_body.dart (hunk 5 above)
Comment (droplet-js) [reply lib/src/multipart_body.dart]: > 

_contentType contains dirty boundary

**Pull request merged**